catalog/ui/search-ui/simple/src/main/java/org/codice/ddf/ui/searchui/simple/properties/UiConfigurationPropertiesFactory.java

```diff
@@ -15,84 +15,48 @@
 
 import java.util.Optional;
 import org.codice.ddf.branding.BrandingRegistry;
+import org.osgi.service.cm.ConfigurationAdmin;
 
 /** Stores external configuration properties. */
-public class ConfigurationStore {
+public class UiConfigurationPropertiesFactory {
 
-  private static ConfigurationStore uniqueInstance;
+  private static final String CONFIG_PID = "ddf.platform.ui.config";
 
-  private String header = "";
+  private static UiConfigurationPropertiesFactory uniqueInstance;
 
-  private String footer = "";
-
-  private String color = "";
+  private Optional<BrandingRegistry> branding = Optional.empty();
 
-  private String background = "";
+  private ConfigurationAdmin configurationAdmin;
 
-  private Optional<BrandingRegistry> branding = Optional.empty();
+  private UiConfigurationPropertiesFactory() {}
 
-  private ConfigurationStore() {
-    header = "";
-    footer = "";
-    color = "";
-    background = "";
+  /**
+   * @return a new UiConfigurationProperties object that contains the properties to build out this
+   *     UI
+   */
+  public UiConfigurationProperties getProperties() {
+    return new UiConfigurationProperties(configurationAdmin, CONFIG_PID, branding);
   }
 
-  /** @return a unique instance of {@link ConfigurationStore} */
-  public static synchronized ConfigurationStore getInstance() {
+  /** @return a unique instance of {@link UiConfigurationPropertiesFactory} */
+  public static synchronized UiConfigurationPropertiesFactory getInstance() {
```

Comment:
❓ Would it be easier to simply create the `UiConfigurationPropertiesFactory` where we declare the `uniqueInstance` field (L25) instead of doing lazy creation and having to make the code synchronized?

Reply:
Yeah, I think so, I'm not sure why it was like this before. I'm guessing it was to make it so this only existed if the configuration existed.